remixer_core/src/main/java/com/google/android/libraries/remixer/Remixer.java

```diff
@@ -77,47 +99,28 @@ public static Remixer getInstance() {
   public void addItem(RemixerItem remixerItem) {
     List<RemixerItem> listForKey = getItemsWithKey(remixerItem.getKey());
     List<RemixerItem> itemsToRemove = new ArrayList<>();
+    Object parentObject = remixerItem.getParentObject();
     for (RemixerItem existingItem : listForKey) {
-      existingItem.assertIsCompatibleWith(remixerItem);
-      if (!existingItem.hasParentObject()) {
-        // The parent activity has been reclaimed by the OS already. It has no callback and it's
-        // still around for keeping the value in sync, saving and checking consistency across types.
-        // Since we're adding a new item that has been asserted to be compatible, it is not
-        // necessary to keep this instance around.
-        itemsToRemove.add(existingItem);
-      } else {
-        // The parent activity is still alive and kicking.
-        if (existingItem.isParentObject(remixerItem.getParentObject())) {
-          // An object with the same key for the same parent object, this shouldn't happen so throw
-          // an exception.
-          throw new DuplicateKeyException(
-              String.format(
-                  Locale.getDefault(),
-                  "Duplicate key %s being used in class %s",
-                  remixerItem.getKey(),
-                  remixerItem.getParentObject().getClass().getCanonicalName()
-              ));
-        }
+      if (parentObject != null && parentObject == existingItem.getParentObject()) {
+        // An object with the same key for the same parent object, this shouldn't happen so throw
+        // an exception.
+        throw new DuplicateKeyException(
+            String.format(
+                Locale.getDefault(),
+                "Duplicate key %s being used in class %s",
+                remixerItem.getKey(),
+                remixerItem.getParentObject().getClass().getCanonicalName()
+            ));
       }
+
     }
-    for (RemixerItem remove : itemsToRemove) {
-      listForKey.remove(remove);
-      remixerItems.remove(remove);
-    }
-    if (remixerItem instanceof Variable && listForKey.size() > 0) {
-      // Make sure that variables use their current value if it has been modified in another
-      // context.
-      // If any modification has been made in any other context to the value of variables with the
-      // same key, otherVariable will have the newest value.
-      Variable otherVariable = (Variable) listForKey.get(0);
-      // At this point newVariable will have the default value only.
-      Variable newVariable = (Variable) remixerItem;
-      // Make newVariable have the current value found in variables that have already been added to
-      // the Remixer.
-      newVariable.setValueWithoutNotifyingOthers(otherVariable.getSelectedValue());
+    if (synchronizationMechanism != null) {
```

Comment:
Nothing terrible, but data would not be saved/synced and the values across different contexts will be completely independent. You could even use the same keys for different data types in different activities, but nothing would break other than actual syncing or persistence.

In fact, without a synchronization mechanism it would allow for the simplest use of Remixer (I just wanna remix something but not care about keys, or persistence, or many rules). I don't want to encourage that use but not necessarily penalize it either.